src/librustdoc/html/static/js/search.js

```diff
@@ -344,7 +341,11 @@ window.initSearch = function(rawSearchIndex) {
             return MAX_LEV_DISTANCE + 1;
         }
 
-        // Check for type name and type generics (if any).
+        // This function checks if the object (`obj`) matches the given type (`val`) and its
+        // generics (if any).
+        //
+        // Returns a Levenshtein distance to the best match. If there is no match, returns
+        // `MAX_LEV_DISTANCE + 1`.
```

Comment:
Can you define the types of `obj`, `val`, and `literalSearch`? In other places in this file, such parameters are documented using [jsdoc](https://jsdoc.app/about-getting-started.html#adding-documentation-comments-to-your-code) syntax, like so:

```js
/**
 * Executes the query and builds an index of results
 * @param  {[Object]} query      [The user query]
 * @param  {[type]} searchWords  [The list of search words to query
 *                                against]
 * @param  {[type]} filterCrates [Crate to search in if defined]
 * @return {[type]}              [A search index of results]
 */
 ```
 
We don't use jsdoc AFAIK, but it's a good syntax for formalizing information about parameters and return types. Also worth noting: I'm not sure why the types are wrapped in brackets. But I think it's okay to follow local style, and we can do a pass later to bring all jsdoc comments in line with the spec.

Reply:
I'm definitely not a big fan of the syntax but why not. :)

Reply:
Fair enough, but I think a syntax that exists and has tooling built for it is much better than no syntax. :-)

Reply:
No sorry, not possible. It's just impossible to read. I'll add the types but the comment is just not human readable...

Reply:
Do you feel that way even for the properly formatted jsdoc style (from https://jsdoc.app/about-getting-started.html#adding-documentation-comments-to-your-code)?

```
/**
 * Represents a book.
 * @constructor
 * @param {string} title - The title of the book.
 * @param {string} author - The author of the book.
 */
function Book(title, author) {
}
```

Reply:
Oh, that's better! That I can do.